mailbox/backup/src/main/java/org/apache/james/mailbox/backup/zip/MailAccountZipIterator.java

```diff
@@ -0,0 +1,272 @@
+        Optional<Long> sizeO = getSize(entry);
+        Optional<Date> internalDateO = getInternalDate(entry);
+        Optional<Flags> flagsO = getFlags(entry);
+
+        if (!messageIdO.isPresent()) {
+            LOGGER.error("No message id in message entry " + entry.getName());
+            return Optional.empty();
+        }
+        if (!mailboxIdO.isPresent()) {
+            LOGGER.error("No mailbox id in message entry " + entry.getName());
+            return Optional.empty();
+        }
+        if (!sizeO.isPresent()) {
+            LOGGER.error("No size defined in message entry " + entry.getName());
+            return Optional.empty();
+        }
+        if (!internalDateO.isPresent()) {
+            LOGGER.error("No internal date defined in message entry " + entry.getName());
+            return Optional.empty();
+        }
+        if (!flagsO.isPresent()) {
+            LOGGER.error("No flags defined in message entry " + entry.getName());
+            return Optional.empty();
+        }
+        InputStream messageContentStream = null;
+        try {
+            messageContentStream = currentContent.asByteSource().openStream();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return Optional.of(new MessageArchiveEntry(messageIdO.get(), mailboxIdO.get(), sizeO.get(), internalDateO.get(), flagsO.get(), messageContentStream));
+    }
+
+
+    private byte[] getCurrentEntryContentBytes() {
+        try {
+            return currentContent.asByteSource().read();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private Optional<String> getAnnotationContent() {
+        byte[] raw = getCurrentEntryContentBytes();
+        if (raw == null) {
+            return Optional.empty();
+        } else {
+            String content = new String(raw, Charsets.UTF_8);
+            return Optional.of(content);
+        }
+    }
+
+    private MailboxAnnotationKey getAnnotationKey(MailboxWithAnnotationsArchiveEntry mailbox, ZipEntry entry) {
+        String key = entry.getName().substring((mailbox.getMailboxName() + "/" + Zipper.ANNOTATION_DIRECTORY + "/").length());
+        return new MailboxAnnotationKey(key);
+    }
+
+    private MailArchiveEntry from(ZipEntry current, ZipEntryType currentEntryType, Optional<ZipEntry> next) throws ZipException {
+        switch (currentEntryType) {
+            case MAILBOX:
+                return fromMailboxEntry(current, next);
+            case MAILBOX_ANNOTATION_DIR:
+                return next();
+            case MAILBOX_ANNOTATION:
+                return fromMailboxAnnotationEntry(current, next);
+            case MESSAGE:
+                Optional<MessageArchiveEntry> messageEntry = fromMessageEntry(current);
+                if (messageEntry.isPresent()) {
+                    return messageEntry.get();
+                }
+        }
+        return null;
```

Comment:
When can this happen? This might not be the most appropriate way to tell the caller we don't know how to handle his input...